packages/k8s.contrib.crd/internal/ModuleGenerator.pkl

```diff
@@ -318,45 +311,76 @@ local function pathPrefixes(path: List<String>): List<List<String>> = new Listin
+      when (applied.additionalProperties != null) {
+        additionalProperties =  mapSchema(applied.additionalProperties!!, (path) { "additionalProperties" }, f)
       }
-    } else mapSchema(applied.items as JsonSchema.Schema, (path) { "items" }, f)
-  }
+      when (applied.items != null) {
+        when (applied.items is Listing<JsonSchema>) {
+          items = new {
+            for (i, item in applied.items as Listing<JsonSchema.Schema>) {
+              [i] = mapSchema(item, (path) { i.toString() }, f)
+            }
+          }
+        } else {
+          items = mapSchema(applied.items as JsonSchema.Schema, (path) { "items" }, f)
+        }
+      }
+    }
 
 local convertedSchemas: Map<JsonSchema, ImportAndType> =
-  utils._findMatchingSubSchemas(filteredRootSchema, List(), (elem) -> elem != filteredRootSchema)
+  utils.findMatchingSubschemas(filteredRootSchema, (elem) -> elem != filteredRootSchema)
     .filter((path, _) -> converters.containsKey(if (logPaths) trace(path) else path))
     .entries
     .fold(Map(), (accumulator: Map<JsonSchema, ImportAndType>, pair) ->
-    let (path = pair.first)
+      let (path = pair.first)
       let (schema = pair.second)
-        let (converted = converters[path])
-          let (reflected = if (converted is Module) reflect.Module(converted) else if (converted is TypeAlias) reflect.TypeAlias(converted) else reflect.Class(converted))
-            accumulator.put(schema, new ImportAndType {
-              type {
-                moduleName = if (reflected is reflect.Module) module.moduleName else reflected.enclosingDeclaration.name
-                name = reflected.name.split(".").last
-              }
-              _import = if (reflected is reflect.Module) reflected.uri else reflected.enclosingDeclaration.uri
-            })
+      let (converted = converters[path])
+      let (reflected =
+        if (converted is Module)
+          reflect.Module(converted)
+        else if (converted is TypeAlias)
+          reflect.TypeAlias(converted)
+        else
+          reflect.Class(converted)
+      )
+        accumulator.put(schema, new ImportAndType {
+          type {
+            moduleName = if (reflected is reflect.Module) module.moduleName else reflected.enclosingDeclaration.name
+            name = reflected.name.split(".").last
+          }
+          local uri =
+            if (reflected is reflect.Module)
+              reflected.uri
+            else
+              reflected.enclosingDeclaration.uri
+          _import = determineImportPath(uri)
+        })
+  )
+
+/// If any conversions are from the core k8s library, replace the import path with [k8sImportPath]
+local function determineImportPath(uri: String): String =
+  if (
+    uri.startsWith("package://pkg.pkl-lang.org/pkl-k8s/k8s")
+    || uri.startsWith("projectpackage://pkg.pkl-lang.org/pkl-k8s/k8s")
   )
+    k8sImportPath + uri.dropWhile((it) -> it != "#").drop(1)
+  else uri
```

Comment:
Added this so that you can use any version of our k8s library, and we'll still replace it with `k8sImportPath`.